app/src/main/java/com/github/siela1915/bootcamp/UploadingRecipeFragment.java

```diff
@@ -124,35 +127,166 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+
+        imgView.setOnClickListener(v -> chooseImg());
+
+        uploadImg.setOnClickListener(v -> {
+            if (isInputValid()) openRecipeReviewDialog();
+            else
+                Toast.makeText(getActivity(), "Please fill required fields before uploading", Toast.LENGTH_LONG).show();
+        });
+
+        addIngredient.setOnClickListener(v -> addIngredient());
+
+        addStep.setOnClickListener(v -> addStep());
+
+        recipeName.addTextChangedListener(new TextValidator(recipeName) {
             @Override
-            public void onClick(View v) {
-                chooseImg();
+            public void validate(TextView textView, String text) {
+                if (!isTextValid(text)) textView.setError("Recipe name is required!");
+                isRecipeNameValid = isTextValid(text);
             }
         });
 
-        uploadImg.setOnClickListener(new View.OnClickListener() {
+        cookTime.addTextChangedListener(new TextValidator(cookTime) {
             @Override
-            public void onClick(View v) {
-                openRecipeReviewDialog();
+            public void validate(TextView textView, String text) {
+                if (!isTextValid(text)) textView.setError("Cooking time is required!");
+                else if (!isNumberPositive(text))
+                    textView.setError("Cooking time must be positive!");
+                isCookTimeValid = isTextValid(text) && isNumberPositive(text);
             }
         });
 
-        addIngredient.setOnClickListener(new View.OnClickListener() {
+        prepTime.addTextChangedListener(new TextValidator(prepTime) {
             @Override
-            public void onClick(View v) {
-                addIngredient();
+            public void validate(TextView textView, String text) {
+                if (!isTextValid(text)) textView.setError("Preparation time is required!");
+                else if (!isNumberPositive(text))
+                    textView.setError("Preparation time must be positive!");
+                isPrepTimeValid = isTextValid(text) && isNumberPositive(text);
             }
         });
 
-        addStep.setOnClickListener(new View.OnClickListener() {
+        servings.addTextChangedListener(new TextValidator(servings) {
             @Override
-            public void onClick(View v) {
-                addStep();
+            public void validate(TextView textView, String text) {
+                if (!isTextValid(text)) textView.setError("Number of serving is required!");
+                else if (!isNumberPositive(text))
+                    textView.setError("Number of serving must be positive!");
+                isServingsValid = isTextValid(text) && isNumberPositive(text);
             }
         });
 
-        return view;
+        addIngredientValidators(ingredientsAmount, ingredientsUnit, ingredientsName);
+        addStepValidators(stepContent);
+
+        addCuisineType.setOnClickListener(v -> {
```

Comment:
Setting the onClickListeners has some repeated code, maybe it would be better to modularize some parts of it